editor/src/clj/editor/grid.clj

```diff
@@ -13,192 +13,385 @@
+                              (cancel-fn nil))))]
+    (doto text-field
+      (ui/text! color)
+      (ui/on-action! update-fn)
+      (ui/on-cancel! cancel-fn)
+      (ui/auto-commit! update-fn))
+    [label text-field]))
+
+(defn- axis-group
+  [app-view prefs prefs-path axis]
+  (let [text-field (TextField.)
+        label (Label. (string/upper-case (name axis)))
+        size-val (str (get (prefs/get prefs prefs-path) axis))
+        cancel-fn (fn [_] (ui/text! text-field size-val))
+        update-fn (fn [_] (try
+                            (let [value (some-> (.getText text-field) Integer/parseInt)]
+                              (if (pos? value)
+                                (do (prefs/set! prefs (conj prefs-path axis) value)
+                                    (ui/text! text-field (str value))
+                                    (invalidate-grids! app-view))
+                                (cancel-fn nil)))
+                            (catch Exception _e
+                              (cancel-fn nil))))]
+    (doto text-field
+      (ui/text! size-val)
+      (ui/on-action! update-fn)
+      (ui/on-cancel! cancel-fn)
+      (ui/auto-commit! update-fn))
+
+    [label text-field]))
+
+(defmethod settings-row :size
+  [app-view prefs option]
+  (let [prefs-path (conj grid-prefs-path option)]
+    (into []
+          (comp (map (partial axis-group app-view prefs prefs-path))
+                (mapcat identity))
+          axes)))
+
+(defn- reset-button
+  [app-view prefs ^PopupControl popup]
+  (let [button (doto (javafx.scene.control.Button. "Reset to defaults")
+                 (.setPrefWidth Double/MAX_VALUE))
+        reset-fn (fn [_]
+                   (doseq [path [[:size :x]
+                                 [:size :y]
+                                 [:size :z]
+                                 [:active-plane]
+                                 [:opacity]
+                                 [:color]]]
+                     (let [path (into grid-prefs-path path)]
+                       (prefs/set! prefs path (:default (prefs/schema prefs path)))))
+                   (invalidate-grids! app-view)
+                   (.hide popup))]
+    (ui/on-action! button reset-fn)
+    button))
+
+(defn- settings
+  [app-view prefs popup]
+  (let [scene-view-id (g/node-value app-view :active-view)
+        grid (g/node-value scene-view-id :grid)
+        options (g/node-value grid :options)
+        reset-btn (reset-button app-view prefs popup)]
+    (->> [:size :active-plane :color :opacity]
+         (e/remove (partial contains? options))
+         (reduce (fn [rows option]
+                   (conj rows (doto (HBox. 5 (ui/node-array (settings-row app-view prefs option)))
+                                (.setAlignment Pos/CENTER))))
+                 [reset-btn]))))
+
+(defn- pref-popup-position
+  ^Point2D [^Parent container]
+  (Utils/pointRelativeTo container 0 0 HPos/RIGHT VPos/BOTTOM -190.0 10.0 true))
```

Comment:
Not sure if it's worth trying to avoid the hardcoded `190` here, since I already had to set a fixed `pref-width` for the popup.